### ISSUE TITLE
Nice to have: "Open with Studio" button for Markdown files.

### ISSUE THREAD

**Issue by smoya** [asyncapi/studio#218]
#### Reason/Context
I'm lately advocating for using Studio whenever I want to add a link to an AsyncAPI file in a Markdown file such as Readmes. Thanks to the `?url=<url-of-file>` query param, Studio can load most of files (yes, not all of them, see https://github.com/asyncapi/studio/issues/127).

I think it would be awesome to create a new design for a button that says "Open with Studio", with the AsyncAPI logo on the left or similar.

#### Description

To design a button that says something like "Open with Studio", with the AsyncAPI logo somewhere that people can use for linking to AsyncAPI files in their Markdown/HML files.

Some examples:

- ![](https://run.pstmn.io/button.svg)
- ![](https://www.deploytodo.com/do-btn-blue.svg)
- ![](https://www.netlify.com/img/deploy/button.svg)

**Comment by alequetzalli** [comment]
I haven't really worried (yet 😈 )  about stuff outside of AsyncAPI's docs and community building, so I haven't been majorly affected by the absence of this.

That said, I can say that I imagine that if I started developing more with our spec, that I would appreciate having **the option** to open an AsyncAPI file in Studio if it empowers me in visualizing events better. 

I think that is a long way to say yes? 😄

**Comment by mcturco** [comment]
@smoya I think this is a great idea! I can definitely work on an image for the button that we can use in markdown files.

> I'd actually go beyond Markdown files and would provide an HTML widget (iframe).

@fmvilas do you mean we would make an iframe to run the Studio on someone's website?

**Comment by magicmatatjahu** [comment]
> do you mean we would make an iframe to run the Studio on someone's website?

I think people can do that on their own, unless we have configured restrictions on the domain side, but I don't think so :) However, we can prepare a dock and a `postmessages` system so that users can react to changes in the studio. What do you think?

**Comment by mcturco** [comment]
> However, we can prepare a dock and a postmessages system so that users can react to changes in the Studio

I think this would be a great feature to add! And maybe if it gets too complicated in the iframe we can send them to the studio website to explore more features

**Comment by fmvilas** [comment]
> @fmvilas do you mean we would make an iframe to run the Studio on someone's website?

No, I meant the button itself is an iframe so when it renders on someone's website we can track it's been used, where, etc.

**Comment by mcturco** [comment]
> No, I meant the button itself is an iframe so when it renders on someone's website we can track it's been used, where, etc.

@fmvilas oh!! yes that's also a great idea and we should do that 👍

**Comment by boyney123** [comment]
Just come to say love the idea @smoya think it can be useful for sure!

**Comment by boyney123** [comment]
@mcturco any ideas for a design for the button we could use?

**Comment by boyney123** [comment]
If we went the iframe solution anybody got any ideas on how this could work?

I'm guessing

- Iframe with a button inside and image 
- Button goes to function/URL somewhere that can track things (maybe in Google Analytics)?
- URL then directs to the studio...

Kind of a middleman/gateway pattern.....

---

**OR**

Also, we could use `IMG-shields` here is an example 👇

**With logo** (we can add ours here https://simpleicons.org/ to use it on the shield)
[![AsyncAPI Open In Studio](https://img.shields.io/badge/AsyncAPI-Open--in--Studio-red?logo=gitpod)](https://studio.asyncapi.com?ref=github)

**Without Logo**
[![AsyncAPI Open In Studio](https://img.shields.io/badge/AsyncAPI-Open--in--Studio-red)](https://studio.asyncapi.com?ref=github)

Using some query param `ref` or something, we could track who is coming from github etc? What kind of metrics are we after here @fmvilas?

**Comment by fmvilas** [comment]
> If we went the iframe solution anybody got any ideas on how this could work?

Sorry, on Markdown you can't actually embed iframe as far as I know. I meant to have an iframe version for HTML but the one for Markdown would have to be an image 👍

**Comment by fmvilas** [comment]
I think the metrics I would be more interested in are:

* Which websites are using the button (feasible by inspecting the HTTP headers when the website is requesting the image)
* How many people are clicking on the buttons (hard to track in Markdown)

Anything else I'm missing? Maybe we should just give up tracking on Markdown? I don't know.

**Comment by smoya** [comment]
> * How many people are clicking on the buttons (hard to track in Markdown)

What if we provide a "button generator" that can either generate HTML or rather Markdown code? We could control the final URL so the tracking might be possible. Of course people could still go in their own creating manually the button, but at least we cover it partially.

**Comment by mcturco** [comment]
> @mcturco any ideas for a design for the button we could use?

@boyney123 yes! I think it could be something very simple. I know we have everything drafted for the new branding right now and it is not actually launched yet, but do you think we should use old brand or new brand?

cc: @fmvilas

**Comment by fmvilas** [comment]
Let's go with new brand IMHO. It shouldn't take us long to launch the new brand.

**Comment by mcturco** [comment]
Hi 👋 
Here are some options for the button. Let's vote on the options here with emojis:

![Screen Shot 2022-02-01 at 5 06 36 PM](https://user-images.githubusercontent.com/60163079/152059693-9fcb07f6-15d2-40d4-a9a7-fefd3b4a6e25.png)

Option 1: react with a ❤️ 
Option 2: react with a 🚀 
Option 3: react with a 🎉 

And of course if anyone has any feedback let me know 😄

**Comment by alequetzalli** [comment]
pretty! :)

**Comment by fmvilas** [comment]
@mcturco Love them all! 🙌 Let's make sure they work well on the dark background too.

**Comment by mcturco** [comment]
> Let's make sure they work well on the dark background too.

@fmvilas aha! Good call. Option 2 will not work the best on dark mode. We could always add a white border around the button to give it the separation it needs on dark mode, but still there might not be enough contrast there.

**Comment by smoya** [comment]
I think this is still relevant as some work has been done around it. Can we remove the `stale` label, please? @mcturco @magicmatatjahu

**Comment by mcturco** [comment]
Here are some new options that work on both light mode and dark mode! Looking for feedback 🙏 

![OpenInStudio - Options](https://user-images.githubusercontent.com/60163079/175107021-b728dcbc-a610-448b-85b5-b98f865df19d.png)

**Comment by smoya** [comment]
> Here are some new options that work on both light mode and dark mode! Looking for feedback 🙏
> 
> ![OpenInStudio - Options](https://user-images.githubusercontent.com/60163079/175107021-b728dcbc-a610-448b-85b5-b98f865df19d.png)

I love them! Even though I have a suggestion, I'm gonna say they could stay like this and I will be happy anyway 😅 

I would suggest to try the following: the logo outline (maybe all lines of it) to have the same color as the text. Maybe it won't look pretty nice, but right now I feel is a bit darker than I would expect. Maybe there is another solution for it, or it is just my perspective and no one else feels the same I do.

**Comment by fmvilas** [comment]
Yeah, I think the background could be lighter, even if it means we have to use another text color to increase contrast. Also, @mcturco don't you have the feeling that the button is beveled? I think the inner gradient (the background) is the culprit of that effect 😄

**Comment by nawed2611** [comment]
Hey, can I take on this issue?

**Comment by magicmatatjahu** [comment]
@nawed2611 Sure, but that issue is more related to design that coding, cc @mcturco

**Comment by magicmatatjahu** [comment]
@mcturco Do you have figma/photoshop file that we can reuse with latest design? :) 

Of course the tracking for use is important but maybe let's make this button available and add the rest later? cc @smoya @fmvilas

**Comment by fmvilas** [comment]
Yup, we have an AsyncAPI Figma account: https://www.figma.com/file/VC4E6Mn6yt5bEnMlzPA2FK/Open-in-Studio-Button?node-id=0%3A1&t=PiHl3pGKYkOxwgNC-1.

**Comment by smoya** [comment]
still relevant

**Comment by smoya** [comment]
> @smoya , is this issue available to be worked on. I would like to work on it as a proposal for the mentorship program?. Kindly tell me what you think, I will be glad for a positive response, thank you.

Hi @Sukriti-m! I'm happy to read you are interested in this but proposal. Of course, you can take it but you will need to find a mentor for this. Maybe @Mayaleeeee would be interested? (no pressure, just pinging)

**Comment by Mayaleeeee** [comment]
> > @smoya , is this issue available to be worked on. I would like to work on it as a proposal for the mentorship program?. Kindly tell me what you think, I will be glad for a positive response, thank you.
> 
> Hi @Sukriti-m! I'm happy to read you are interested in this but proposal. Of course, you can take it but you will need to find a mentor for this. Maybe @Mayaleeeee would be interested? (no pressure, just pinging)

Thank you for tagging me. I'm happy to mentor her, @smoya

**Comment by Sukriti-m** [comment]
> > > @smoya , is this issue available to be worked on. I would like to work on it as a proposal for the mentorship program?. Kindly tell me what you think, I will be glad for a positive response, thank you.
> > 
> > 
> > Hi @Sukriti-m! I'm happy to read you are interested in this but proposal. Of course, you can take it but you will need to find a mentor for this. Maybe @Mayaleeeee would be interested? (no pressure, just pinging)
> 
> Thank you for tagging me. I'm happy to mentor her, @smoya

Hey @Mayaleeeee 
I will be glad to have you as my mentor. However, I'd like to ask is this issue sufficient for creating a proposal or do I need to propose more features along with it to get selected for the mentorship program?

**Comment by ayushnau** [comment]
What is this going still not done. if not then i would like to work on this.

**Comment by smoya** [comment]
> What is this going still not done. if not then i would like to work on this.

Please feel free

**Comment by ashmit-coder** [comment]
Hey @smoya I would like to work on this issue is it still relevant?

**Comment by RobinTail** [comment]
Voting for this — I need the `?url=` feature, so that Studio could open the request URL in one click.

**Comment by smoya** [comment]
> Voting for this — I need the `?url=` feature, so that Studio could open the request URL in one click.

That feature is already implemented. See https://studio.asyncapi.com/?url=https://raw.githubusercontent.com/asyncapi/spec/master/examples/streetlights-mqtt-asyncapi.yml
